src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakPickerMaxima.cpp

```diff
@@ -118,15 +118,14 @@ namespace OpenMS
           act_snt_r2 = int_array[i + 2] / noise_estimator.get_noise_value(mz_array[i + 2]);
         }
 
-        if ((i > 1
-            && (!check_spacings || std::fabs(left_neighbor_mz - mz_array[i - 2]) < spacing_difference_ * min_spacing)
+        if (i > 1
+            && (i + 2) < mz_array.size()
             && left_neighbor_int < int_array[i - 2]
-            && act_snt_l2 >= signal_to_noise_)
-           &&
-            ((i + 2) < mz_array.size()
-            && (!check_spacings || std::fabs(mz_array[i + 2] - right_neighbor_mz) < spacing_difference_ * min_spacing)
             && right_neighbor_int < int_array[i + 2]
-            && act_snt_r2 >= signal_to_noise_)
+            && act_snt_l2 >= signal_to_noise_
+            && act_snt_r2 >= signal_to_noise_
+            && (!check_spacings || std::fabs(left_neighbor_mz - mz_array[i - 2]) < spacing_difference_ * min_spacing)
```

Comment:
yes, but it makes reading it quite a bit harder. maybe execution is faster, I dont know